### ISSUE TITLE
Fix accessibility issues found by axe-core

### ISSUE THREAD

**Issue by robertknight** [hypothesis/client#1734]
Adding [runtime accessibility tests](https://github.com/hypothesis/client/pull/1729) using axe-core found 9 failures.

In [the PR](https://github.com/hypothesis/client/pull/1729) these failures have been flagged by `// FIXME-A11Y` comments in the test files followed by a skipped test (`it.skip("should pass a11y checks")`).

These issues include:

- Anchors with no text or label
- Color contrast failures in the tests for `AnnotationDocumentInfo`, `AnnotationQuote`, `AnnotationShareInfo`, `LoggedOutMessage`, `ShareAnnotationsPanel` and `VersionInfo`

**Comment by lyzadanger** [comment]
@robertknight I think this task may be complete. Can you confirm?

**Comment by robertknight** [comment]
Yes, this is complete AFAIK.
